private_transactions/src/error.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::fmt;
+use ethereum_types::Address;
+use trie::TrieError;
+use ethcore::account_provider::SignError;
+use ethcore::error::ExecutionError;
+use transaction::Error as TransactionError;
+use ethkey::Error as KeyError;
+
+#[derive(Debug)]
+/// Errors concerning private transaction processing.
+pub enum PrivateTransactionError {
```

Comment:
use [error_chain](https://crates.io/crates/error-chain). We don't do manual error conversion in parity any more

Reply:
error_chain will also add `std::error::Error` and `std::fmt::Display` error implementations so it will be easier integrate it with external modules